src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -67,7 +68,7 @@ public static void Initialize()
             // Check Instrumentation is already initialized with native profiler.
             Type profilerType = Type.GetType("OpenTelemetry.AutoInstrumentation.Instrumentation, OpenTelemetry.AutoInstrumentation");
 
-            if (profilerType == null)
+            if (profilerType == null && EvaluateOpenTelemetryReference(loaderAssemblyLocation))
```

Comment:
`EvaluateOpenTelemetryReference` returns a boolean it gives the reader the impression that we can hit the else of the if statement due to it. However, it only returns true throwing and exception otherwise. It seem better to change its return type to `void` and move it outside the if expression. The name should reflect that it throws in case of error, my suggestion will be something like `ThrowIfReferenceIncorrectOpenTelemetryVersion`